app/cafe.py

```diff
@@ -0,0 +1,21 @@
+import datetime
+from app.errors import (
+    NotVaccinatedError,
+    OutdatedVaccineError,
+    NotWearingMaskError
+)
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, visitor: dict) -> str:
+        if "vaccine" in visitor:
+            if visitor["vaccine"]["expiration_date"] < datetime.date.today():
+                raise OutdatedVaccineError("Your vaccine is out of date")
+            if not visitor["wearing_a_mask"]:
+                raise NotWearingMaskError("You should wear mask")
+            return f"Welcome to {self.name}"
+        else:
+            raise NotVaccinatedError("All friends should be vaccinated")

```

Comment:
Code Efficiency: Instead of checking if the 'vaccine' key exists in the visitor dictionary and then checking the 'wearing_a_mask' key, you can use the dictionary's get() method. This method returns the value for the given key if it exists, otherwise it returns a default value. This way, you can avoid unnecessary if-else blocks and make your code more efficient. Here is an example:

python
vaccine = visitor.get('vaccine')
wearing_a_mask = visitor.get('wearing_a_mask')

if not vaccine or vaccine['expiration_date'] < datetime.date.today():
    raise OutdatedVaccineError('Your vaccine is out of date')
if not wearing_a_mask:
    raise NotWearingMaskError('You should wear a mask')
```